### PR TITLE
fix(warehouse): Batch G data integrity — dimension validation, verification guards, canonical onboarding stock, quick-audit prompt

### LINKED ISSUE #75: [Warehouse] Audit System — Count + Organization, Confidence, Speed Mode
**Parent:** #56 (Warehouse & Storage System)

### Count Audit Tab
- Queue sorted by lowest-confidence area first
- Walking path order: Area → Shelf → Row
- **System count HIDDEN** during user count (prevents copying)
- Shown AFTER submission with variance
- Speed mode: QR sticker auto-triggers, camera ready for next area
- Misplaced part: add to cart (sort later) OR quick fix (scan correct spot)
- Multi-user consensus: 2-3 users count independently; majority wins

### Organization Audit Tab
- Separate process: monthly/manager-initiated/system-suggested
- Labels accurate? Parts in right spots? Overcrowded? Misplaced?
- Consolidation voting: user suggests best home for duplicate-location parts
- Manager override; 3 ignores → escalate with required reason

### "While You're Here" Quick Audit
- Triggers at area; only <85% confidence parts shown
- Once per area per day; ~10 seconds
- Resets audit timer + decay factor

### 10-Level Reliability Scale
| Level | Confidence Start | Decay to 80% |
|-------|-----------------|---------------|
| 0-2 | 0-50% | N/A or ~3mo |
| 3-4 | 100% → decays | ~3 months |
| 5-6 | 100% → decays | ~4-6 months |
| 7-8 | 100% → decays | ~9-11 months |
| 9 | Slowest decay | ~12 months |
| 10 | Theoretical | Instant to 9 |

### Checklist
- [ ] Count Audit tab with confidence queue
- [ ] System count hidden during counting
- [ ] Speed mode with QR triggers
- [ ] Multi-user consensus
- [ ] Organization Audit tab
- [ ] Consolidation voting
- [ ] "While You're Here" quick audit
- [ ] Confidence decay calculations
- [ ] Movement decay factor
- [ ] Dollar value threshold matching
- [ ] Misplaced part tracking
- [ ] Help button with audit guide

---
🤖 Generated with [Claude Code](https://claude.com/claude-code)

### LINKED ISSUE #91: [Warehouse] Onboarding Wizard — 6-Step Progressive Setup
**Parent:** #56 → #88 (Warehouse Locations)

### 6-Step Guided Setup
1. **Room Setup:** Define warehouse room dimensions (tape measure)
2. **Place Units:** Drag storage units onto floor plan grid
3. **Number System:** Name rows, units, shelves (sticker labels)
4. **Walking Path:** Define walking order for audits
5. **First Count:** Quick count of all parts (no locations needed)
6. **Set Targets:** MIN/TARGET/MAX for key parts

### Progressive Levels
- Level 0: Just counts → Level 5: Full bins
- Can stop at any level and use the system
- Progressively unlocks features as warehouse detail increases

### Alternative: Quick Count
Skip layout entirely → scan/search parts → add counts + optional location

### First-Time Experience
- Friendly intro: "Let's set up your warehouse"
- Estimated time per step
- [Skip for Now] on each step
- Progress indicator (step 2 of 6)
- Celebration on completion

### Checklist
- [ ] 6-step wizard with progress indicator
- [ ] Room dimensions input
- [ ] Drag-drop unit placement
- [ ] Naming/numbering system
- [ ] Walking path definition
- [ ] First count mode
- [ ] Target setting
- [ ] Quick Count alternative
- [ ] [Skip for Now] per step
- [ ] Progressive level tracking

---
🤖 Generated with [Claude Code](https://claude.com/claude-code)

### LINKED ISSUE #494: [GH#486][Warehouse][Audit][Bug] Send-for-verification lacks no-eligible and already-flagged handling
## Environment / context
- Paperclip issue: WEI-1343, GH#486 QA verification lane
- Commit under test: `ef69b490e`
- Scope: `SendForVerificationSheet` default + no eligible users inline error + already flagged toast paths from the GH#486 acceptance evidence list

## Reproduction steps
No-eligible-users path:
1. Use `SendForVerificationSheet` or `QueueSendForVerificationSheet` when fewer than the required eligible active counters exist after excluding the current operator.
2. Tap `Send`.

Already-flagged path:
1. Send a part/session for multi-user verification.
2. Attempt to send the same part/session for verification again.

## Expected result
- No eligible users: sheet stays open and shows the required inline error explaining no eligible counters are available.
- Already flagged: operator gets the expected already-flagged toast/path and duplicate assignments are not created.

## Actual result
The implementation has no distinct handling for either condition:
- `WarehouseService.flagForMultiUserAudit(...)` selects up to the requested users and returns whatever rows it creates, including an empty array when no eligible users exist. It does not throw or report insufficient eligible users.
- The send sheets ignore the returned assignment count, call `onSent()`, and dismiss even if zero assignments were created.
- The service does not check for an existing unresolved assignment for the same part/session before inserting new `MultiUserAuditAssignment` rows, so the already-flagged path can create duplicates instead of surfacing the required toast/error state.

## Evidence
Source check on `ef69b490e`:
- `core/Sources/WiredPartCore/Services/WarehouseService.swift:4807-4882` returns created assignments without no-eligible/duplicate checks.
- `Weird Parts IOS/Weird Parts IOS/Features/Warehouse/SendForVerificationSheet.swift:69-80` ignores the returned assignments and dismisses on success.
- `Weird Parts IOS/Weird Parts IOS/Features/Warehouse/IOSAuditPage.swift:1521-1532` has the same queue-sheet behavior.

## Severity
High for GH#486 QA: two required acceptance evidence paths cannot be honestly passed, and the already-flagged case risks duplicate work assignments.

## Suggested owner
FrontendCoder, with service-level guard plus UI-specific inline/toast handling.

## Duplicate check
Searched GitHub issues before filing:
- `"no eligible users" "already flagged" warehouse verification` returned no matching issue.
- `"SendForVerificationSheet" GH#486` returned no matching issue.

### LINKED ISSUE #1165: [Warehouse][Bug] Storage units can be saved with invalid dimensions
Status: OPEN

## Problem
`WarehouseService.addStorageUnit` accepts non-positive physical dimensions and grid dimensions at the service boundary. The recent #1156 fix added guards for the `createStorageUnit(levels:areasPerLevel:)` convenience helper, but the lower-level `addStorageUnit` path used by the iOS Add Storage Unit sheet still inserts invalid `widthInches`, `depthInches`, `heightInches`, `gridWidth`, and `gridHeight` values directly.

## Reproduction
1. In `core/Sources/WiredPartCore/Services/WarehouseService.swift`, create or use a floor plan.
2. Call `addStorageUnit` with any non-positive dimension, for example:
   - `widthInches: 0`
   - `depthInches: 0`
   - `heightInches: 0`
   - `gridWidth: 0`
   - `gridHeight: 0`
3. The service persists the unit instead of throwing `WarehouseError.invalidDimension`.

A focused adversarial Swift Testing probe confirmed all six invalid cases were accepted:

```swift
#expect(throws: WarehouseService.WarehouseError.invalidDimension) {
    _ = try env.warehouse.addStorageUnit(
        floorPlanId: plan.id!,
        name: "Invalid Unit",
        unitType: "rack",
        widthInches: input.widthInches,
        depthInches: input.depthInches,
        heightInches: input.heightInches,
        gridWidth: input.gridWidth,
        gridHeight: input.gridHeight
    )
}
```

Command run:

```bash
swift test --filter 'BugHunterStorageUnitProbeTests/addStorageUnitRejectsInvalidDimensions'
```

Result: failed with six expectation failures: `Expectation failed: an error was expected but none was thrown`.

## Evidence
- `addStorageUnit` only checks that the name is non-empty before insert: `core/Sources/WiredPartCore/Services/WarehouseService.swift:4801`.
- The recent guard covers only `createStorageUnit` hierarchy counts: `core/Sources/WiredPartCore/Services/WarehouseService.swift:4886`.
- iOS Add Storage Unit sheet calls `addStorageUnit` directly with physical and grid dimensions: `Weird Parts IOS/Weird Parts IOS/Features/Warehouse/WarehouseLocationsPage.swift:997`.
- Related but distinct existing issue: #1151 tracks floor feature grid validation, not storage unit physical/grid dimensions.

## Expected
`addStorageUnit` should reject invalid dimensions at the service layer:
- `widthInches`, `depthInches`, `heightInches` when present must be greater than zero.
- `gridWidth`, `gridHeight` when present must be greater than zero.
- `gridX`, `gridY` when present should not be negative.
- If a floor plan has saved grid bounds, storage unit placement should not extend outside the grid.

## Actual
Invalid storage unit rows can be persisted. UI steppers reduce normal user exposure, but service callers, tests, imports, sync/restore paths, or future automation can create impossible units that later render as zero/negative-size or off-grid warehouse elements.

## Impact
Severity: Medium
Blast radius: Warehouse floor-plan editor, warehouse onboarding/import/sync paths, and downstream AI/QA context that trusts saved storage geometry. This can create corrupted layout data that is hard for a field user to diagnose or repair.

## Suggested fix
Add storage-unit dimension and placement validation to `addStorageUnit`, mirroring the zone/floor-feature validation approach where appropriate. Add regression tests for non-positive physical dimensions, non-positive grid dimensions, negative coordinates, and over-bound placement.

Found during Paperclip WEI-4089 daily exploratory bug hunt.

### PR DESCRIPTION
**Pull request by xXKillerNoobYT**
## Batch G — Warehouse data integrity

Plan: docs/plans/beta-readiness-issue-resolution-2026-07.md §3 Batch G

Closes #1165
Closes #494
Closes #91
Closes #75

### What changed

**#1165 — Storage-unit dimension/placement validation**
- `WarehouseService.addStorageUnit` now rejects non-positive `widthInches` / `depthInches` / `heightInches`, non-positive `gridWidth` / `gridHeight`, negative `gridX` / `gridY`, and placement outside saved floor-plan grid bounds (`WarehouseError.invalidDimension`), mirroring the zone/floor-feature validation approach.
- `updateStorageUnit` gets the same guards; the floor-plan bounds check only runs when a placement field is actually changing so unrelated updates (rename, configure) never trip on legacy rows.
- Add/Edit Storage Unit sheets surface the rejection inline with a specific message (new `invalidDimension` mapping in `userFriendlyError`), fixed wrong "load locations" error contexts.

**#494 — Send-for-verification guards**
- `flagForMultiUserAudit` now throws `partAlreadyFlaggedForVerification` when the part already has unresolved (pending/counted) assignments for the same session, and `noEligibleVerificationCounters(required:available:)` when fewer eligible active counters exist than requested (minimum 2 for consensus) — instead of silently creating zero or duplicate assignments.
- `QueueSendForVerificationSheet` surfaces both errors inline (distinct messages via `userFriendlyError`) and stays open; added `.presentationDetents` to the sheet per repo convention.

**#91 — PartsFlowWizard canonical persistence (last open slice per beta plan)**
- New `WarehouseService.recordPartsFirstSetupCount(partId:countedQty:performedBy:)` upserts the canonical warehouse `stock` row (`warehouse`/1) and records a signed adjustment `stock_movements` row when quantity changes.
- `PartsFlowWizard.saveAllProgress` now persists counts via that method and locations to the canonical `parts.shelf_location` field, replacing the old "Location: X | Initial count: N" notes-field concatenation (which also clobbered user notes).

**#75 — "While You're Here" quick-audit prompt (last open slice)**
- New `WarehouseService.getQuickAuditCandidates(areaId:limit:)` returns parts at an area with confidence ≤ 85% (the quick-verification trigger), lowest first.
- The dashboard QR scanner now offers the prompt after scanning a warehouse location whose parts include low-confidence entries: "N parts here need a quick count." with Count Now / Not Now, at most once per area per day (`QuickAuditPromptStore`, per-user in UserDefaults).
- New `QuickAuditCountSheet` records counts via the existing `recordQuickVerificationCount` (resets confidence to 100% and the movement decay factor). System counts stay hidden while counting, per the audit spec.

### Test evidence

- `cd core && swift build` → Build complete.
- `cd core && swift test --filter "WarehouseAuditTests|WarehouseFloorPlanTests"` → 116 tests in 2 suites passed (includes 11 new tests: dimension/placement validation, no-eligible + already-flagged + re-flag-after-resolution guards, quick-audit candidates, parts-first setup count).
- `cd core && swift test --filter "E2EWarehouseTests|WarehouseWalkingPathTests|WarehouseLocationTests|WarehouseServiceExtTests"` → 178 tests in 4 suites passed.
- `cd core && swift test` (full suite) → 2269 tests in 68 suites passed (139.97s), 0 failures.
- `xcodebuild -workspace "Weird Parts.xcworkspace" -scheme "WiredPart-iOS" -destination "generic/platform=iOS Simulator" build CODE_SIGNING_ALLOWED=NO` → succeeded (exit 0); only pre-existing warnings in untouched files.
- `xcrun swiftc -parse` on all 5 touched app-target files → clean.

### Notes
- #91 and #75 are umbrella trackers; per the beta plan (§3 Batch G) these were the last open slices, so both carry Closes lines. Earlier triage lanes (WEI-1090/1092 progress-persistence and zone-UX spec; #486 operator submission flow) were addressed or tracked separately per the plan.
- `updateFloorPlanGrid`'s shrink-orphan check (#1240) still only counts features/zones, not storage units — pre-existing scope, unchanged here.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by xXKillerNoobYT** [comment]
Review finding addressed in 406552753 — dead-end multi-user verification sets are now recoverable.

**Problem (important):** the new duplicate-flag guard blocked any `pending`/`counted` set. When 2 counters disagreed, `resolveMultiUserAudit` returned nil and left rows `counted`; re-flagging (the pre-PR retry path) then threw `partAlreadyFlaggedForVerification` forever, and the error message told users to wait for a resolution that could never happen. Session-less sets (QueueSendForVerificationSheet with no active session) had no resolution path at all, permanently blocking future no-session flags for that part.

**Fix:**
- Guard only throws while verification is genuinely in flight: counters still pending, or submitted counts that can reach consensus (session sets) and just await resolution.
- Fully-counted dead-end sets — disagreeing counts, or any fully-counted session-less set — are marked `superseded` and the re-flag proceeds with a fresh assignment set (history preserved).
- `resolveMultiUserAudit` and `getMultiUserAuditAssignments` skip `superseded` rows so a dead set never pollutes its replacement's consensus or the summary UI. No schema change needed (`status` has no CHECK constraint; `getMyMultiUserAuditAssignments` already filters `status = 'pending'`).
- User-facing message updated: "Wait for the counts to come in, or resolve the submitted counts, before sending it again." — now always actionable.

**Test evidence:**
- `swift test --filter WarehouseAuditTests` — 67 tests passed, including 3 new regression tests (supersede-on-disagreement + replacement set resolves cleanly, block-while-consensus-reachable, session-less dead-end re-flag).
- `swift test --filter "Warehouse"` — 303 tests in 9 suites passed.
- `swift build` — Build complete.
- `xcrun swiftc -parse "Weird Parts IOS/Weird Parts IOS/Shared/UserFriendlyError.swift"` — exit 0.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

File: core/Sources/WiredPartCore/Services/WarehouseService.swift [hunk 1]
```diff
@@ -2766,6 +2774,73 @@ public final class WarehouseService: Sendable {
...
+                VALUES (?, ?, 'warehouse', 1, 'warehouse', 1,
+                        '\(StockMovement.MovementType.adjustment.rawValue)', ?, ?, ?, datetime('now'))
+                """, arguments: [
+                    partId,
+                    delta,
+                    "Parts-first setup count",
+                    "Initial count: \(currentQty ?? 0) -> \(countedQty) (\(signedDelta))",
+                    performedBy,
+                ])
+            return delta
```
[the remark this answers is not in the document]
Comment (xXKillerNoobYT) [reply core/Sources/WiredPartCore/Services/WarehouseService.swift:2842]: Confirmed: performedBy was optional but stock_movements.performed_by is NOT NULL, so an omitted user id would fail the insert at runtime after the stock row was already upserted. Fixed in 7c5caf0 by making performedBy: Int64 required and validating it with the same requireActiveUser gate every other WarehouseService movement method already uses; updated the one call site (PartsFlowWizard) and the validation test accordingly. All 67 WarehouseAuditTests pass.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: Weird Parts IOS/Weird Parts IOS/Features/Warehouse/PartsFlowWizard.swift [hunk 2]
```diff
@@ -564,7 +564,8 @@ struct PartsFlowWizard: View {
         guard validateBeforeSaving() else { return }
         isSaving = true
 
-        guard let service = appCore.partsService else {
+        guard let service = appCore.partsService,
+              let warehouseService = appCore.warehouseService else {
             isSaving = false
             saveErrorMessage = "Parts service unavailable. Your draft is still saved on this device."
```
[the remark this answers is not in the document]
Comment (xXKillerNoobYT) [reply Weird Parts IOS/Weird Parts IOS/Features/Warehouse/PartsFlowWizard.swift]: Confirmed: the guard covers both partsService and warehouseService being unavailable, but the message only mentioned parts. Fixed in 7c5caf0 — the message now reads "Parts or warehouse service unavailable. Your draft is still saved on this device."

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: Weird Parts IOS/Weird Parts IOS/Shared/UserFriendlyError.swift [hunk 3]
```diff
@@ -23,13 +23,29 @@ func userFriendlyError(_ error: Error, context: String = "load data") -> String
...
+        case .gridShrinkWouldOrphanItems(let features, let zones):
+            // Fixed counts only — no row contents (see security note above).
+            var stranded: [String] = []
+            if features > 0 { stranded.append("\(features) feature\(features == 1 ? "" : "s")") }
+            if zones > 0 { stranded.append("\(zones) zone\(zones == 1 ? "" : "s")") }
+            return "Can't shrink the grid: \(stranded.joined(separator: " and ")) would fall outside the new size. Move or remove them first."
+        case .invalidDimension:
+            // Issue #1165: service-layer dimension/placement validation.
+            // Fixed string — no dimension values from the failed input.
+            return "Dimensions and grid placement must be positive and fit inside the floor-plan grid."
```
[the remark this answers is not in the document]
Comment (xXKillerNoobYT) [reply Weird Parts IOS/Weird Parts IOS/Shared/UserFriendlyError.swift]: Confirmed: WarehouseService validation only rejects gridX/gridY < 0 (0 is valid); only width/height/depth must be strictly positive. Fixed in 7c5caf0 — the .invalidDimension message now reads "Dimensions must be positive, and grid placement must be zero or greater, and fit inside the floor-plan grid."

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_
